src/gstools/field/generator.py

```diff
@@ -548,3 +559,302 @@ def _create_unit_vector(self, broadcast_shape, axis=0):
+        * :math:`k_j` : samples from the spectral density distribution of
+          the covariance model
+        * :math:`W_{k,j}` : cross-product of the random normal vector Z_{k,j} with the wave-vector k_j
+          ensuring the incompressibility
+
+    References
+    ----------
+    .. [Kraichnan1970] Kraichnan, R. H.,
+           "Diffusion by a random velocity field.",
+           The physics of fluids, 13(1), 22-31., (1970)
+    """
+
+    def __init__(
+        self,
+        model,
+        mean_velocity=1.0,
+        mode_no=1000,
+        vec_dim=None,
+        seed=None,
+        verbose=False,
+        sampling="auto",
+        periodic_bc=False,
+        box_len=None,
+        **kwargs,
+    ):
+        if vec_dim is None and (model.dim < 2 or model.dim > 3):
+            raise ValueError(
+                "Only 2D and 3D incompressible vectors can be generated."
+            )
+        if vec_dim is not None and (vec_dim < 2 or vec_dim > 3):
+            raise ValueError(
+                "Only 2D and 3D incompressible vectors can be generated."
+            )
+        super().__init__(model, mode_no, seed, verbose, sampling, **kwargs)
+
+        self.mean_u = mean_velocity
+        if vec_dim is None:
+            self.vec_dim = model.dim
+        else:
+            self.vec_dim = vec_dim
+        self._value_type = "vector"
+        
+        self.periodic_bc=periodic_bc
+        self.box_len=box_len
+
+        # for using the Kraichnan method for zero-velocity fluid, z_1 and z_2 must instead contain N=self._mode_no independent realizations of normal vectors of size vec_dim,
+        mean = np.zeros(self.vec_dim)
+        cov = np.identity(self.vec_dim)
+
+        self._z_1 = self._rng.random.multivariate_normal(mean, cov, size=self._mode_no) # shape (_mode_no, self.vec_dim)
+        self._z_2 = self._rng.random.multivariate_normal(mean, cov, size=self._mode_no)
+        print("shape of z_1: ",np.shape(self._z_1))
+        
+    def __call__(self, pos):
+        """Calculate the random modes for the randomization method.
+
+        This method  calls the `summate_incompr_*` Cython methods,
+        which are the heart of the randomization method.
+        In this class the method contains a projector to
+        ensure the incompressibility of the vector field.
+
+        Parameters
+        ----------
+        pos : (d, n), :class:`numpy.ndarray`
+            the position tuple with d dimensions and n points.
+
+        Returns
+        -------
+        :class:`numpy.ndarray`
+            the random modes
+        """
+        pos = np.asarray(pos, dtype=np.double)
+        
+        print("\nStarting summate_incompr_zero_vel")
+
+        if self.periodic_bc:
+            if not self.box_len==None:
+                print("\nimposing periodic boundary conditions on spatial coordinates!")
+                print("\nrounding first 'vec_dim' vector components in cov_sample to multiples of 2pi/box_length")
```

Comment:
I think these are nice debug prints, but for production code, they should be removed or at least you should use the `verbose` keyword, as is done in elsewhere in this file.